docs/resource-cleanup.md

```diff
@@ -0,0 +1,85 @@
+# Stale Resource Cleanup
+
+If the PVC is created with storage class which is having the `reclaimPolicy``
+as `Retain` will not delete the PV object , backend omap metadata and backend image.
+Manual deletion of PV will result in stale omap keys, values , cephfs subvolume and rbd image.
+It is required to cleanup metadata and image separately.
+
+## Steps
+
+
+### 1. Get volume ID from PV
+
+a. get pv_name
+
+  `[$]kubectl get pvc pvc_name -n namespace -owide`
+  ```
+  $ kubectl get pvc mysql-pvc -owide -n prometheus
+  NAME        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE   VOLUMEMODE
+  mysql-pvc   Bound    pvc-bc537af8-67fc-4963-99c4-f40b3401686a   20Gi       RWO            csi-rbd        14d   Filesystem
+  ```
+
+### 2. Get omap keys
+
+a. get omapkey_pvc (suffix of csi.volumes.default is value used for the CLI option [--instanceid](deploy-rbd.md#configuration) in the provisioner (or, nodeplugin daemonset) deployment.)
+
+  `[$]rados listomapkeys csi.volumes.default -p pool_name | grep pv_name`
+  ```
+  $ rados listomapkeys csi.volumes.default -p kube_csi | grep pvc-bc537af8-67fc-4963-99c4-f40b3401686a
+  csi.volume.pvc-bc537af8-67fc-4963-99c4-f40b3401686a
+  ```
+b. get omapkey
+
+  `[$]rados getomapval csi.volumes.default omapkey_pvc -p pool_name`
+  ```
+  $ rados getomapval csi.volumes.default csi.volume.pvc-bc537af8-67fc-4963-99c4-f40b3401686a -p kube_csi
+  value (36 bytes) :
+  00000000  64 64 32 34 37 33 64 30  2d 36 61 38 63 2d 31 31  |dd2473d0-6a8c-11|
+  00000010  65 61 2d 39 31 31 33 2d  30 61 64 35 39 64 39 39  |ea-9113-0ad59d99|
+  00000020  35 63 65 37                                       |5ce7|
+  00000024
+  ```
+
+
+### 3. Delete the RBD image or CephFS subvolume
```

Comment:
this can move to 2 option and `Get omap keys` can be moved to 3

Reply:
According to the document we get image name or subvolume name from omapvalue so I think getting omap should come first .

Reply:
correct, sorry i missed that